src/gmail.js

```diff
@@ -961,6 +961,26 @@ var Gmail = function(localJQuery) {
             if (pathname && (pathname.endsWith("/i/s") || pathname.endsWith("/i/fd"))) {
                 api.tools.parse_request_payload(params, triggered);
             }
+            if (pathname && (pathname.endsWith("/i/s"))) {
+                if (params.body_params && params.body_params[2] && params.body_params[2][1] && params.body_params[2][1][0][2][2][7]) {
+                    // check params length  and get thread ids
```

Comment:
My experience trying to access specific indices on the request payload without further analysis has been less than impressive. For my use-case (sent email), it has often resulted in unreliable detection of events and meta-data.

For that reason I created the `api.tools.extract_from_graph()` function and associated graph-analysis-functions like `api.check.data.is_thread()` and `api.check.data.is_email()` (all of which are easily unit-tested).

Reading this code, it's not clear to me what kind structure we are looking for, and what kind of graph-entity it "is".

Would it be possible to rewrite this code to be more in line with how `api.tools.parse_request_payload()` already does things?

Together with that, if you have a screenshot of an example payload like this (to help illustrate what kind of request we're processing here) it will make it *much* easier to comment on the parsing code itself too.